itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -1504,12 +1504,20 @@ public void testDumpExternalTableSetFalse() throws Throwable {
             .run("use " + replicatedDbName)
             .run("show tables like 't3'")
             .verifyResult("t3")
+            .verifyExternalTable(replicatedDbName, "t3",
+                    primary.getTableLocation(primaryDbName, "t3"))
             .run("select id from t3 where id = 10")
-            .verifyFailure(new String[] {"10"});
+            // Since both the external tables point to the same location as they share the same
+            // file system, the data in external table on the primary is also visible through the
```

Comment:
Since the test is designed to have primary and replica on the same DFS, when we create an external table on replica with same location as primary, the data automatically shows up. In a reply to another comment I have explained why do we want to have the same location for external tables on source as well as the target. If we modify the test to use different DFS for primary and replica, we won't need this change.